src/ILLink.Shared/ILLink.LinkAttributes.xsd

```diff
@@ -0,0 +1,161 @@
+			</xsd:complexContent>
+		</xsd:complexType>
+	</xsd:element>
+	
+	<xsd:element name="field">
+		<xsd:complexType>
+			<xsd:complexContent>
+				<xsd:extension base="typechild" />
+			</xsd:complexContent>
+		</xsd:complexType>
+	</xsd:element>
+	
+	<xsd:element name="property">
+		<xsd:complexType>
+			<xsd:complexContent>
+				<xsd:extension base="typechild" />
+			</xsd:complexContent>
+		</xsd:complexType>
+	</xsd:element>
+
+	<xsd:element name="argument">
+		<xsd:complexType mixed="true">
+			<xsd:choice minOccurs="0" maxOccurs="1">
+				<xsd:element name="argument" type="xsd:string"/>
+			</xsd:choice>
+			<xsd:attribute name="type" />
+		</xsd:complexType>
+	</xsd:element>
+	
+	<xsd:element name="attribute">
+		<xsd:complexType>
+			<xsd:choice maxOccurs="unbounded" minOccurs="0">
+				<xsd:element ref="argument" />
+				<xsd:element minOccurs="0" maxOccurs="unbounded" name="property">
+					<xsd:complexType mixed="true">
+						<xsd:choice maxOccurs="unbounded" minOccurs="0">
+							<xsd:element ref="attribute"></xsd:element>
+						</xsd:choice>
+						<xsd:attribute name="name" />
+					</xsd:complexType>
+				</xsd:element>
+			</xsd:choice>
+			<xsd:attribute name="internal" type="xsd:string" />
+			<xsd:attribute name="fullname" type="xsd:string" />
+			<xsd:attribute name="assembly" type="xsd:string" />
+		</xsd:complexType>
+	</xsd:element>
+
+	<xsd:element name="namespace">
+		<xsd:complexType>
+			<xsd:attribute name="fullname"/>
+		</xsd:complexType>
+	</xsd:element>
+	
+	<xsd:element name="assembly">
+		<xsd:complexType>
+			<xsd:choice maxOccurs="unbounded" minOccurs="0">
+				<xsd:element ref="attribute" />
+				<xsd:element ref="type" />
+				<xsd:element ref="namespace" />
+			</xsd:choice>
+			<xsd:attribute name="fullname" type="xsd:string" use="required" />
+			<xsd:attribute name="name" />
+			<xsd:attribute name="signature" />
+			<xsd:attribute name="feature" />
+			<xsd:attribute name="featurevalue" type="xsd:boolean" />
+			<xsd:attribute name="featuredefault" type="xsd:boolean" />
+		</xsd:complexType>
+	</xsd:element>
+	
+	<xsd:element name="type">
+		<xsd:complexType>
+			<xsd:choice maxOccurs="unbounded" minOccurs="0">
+				<xsd:element ref="attribute"></xsd:element>
+				<xsd:element ref="event" />
+				<xsd:element ref="method" />
+				<xsd:element ref="field" />
+				<xsd:element ref="property" />
+				<xsd:element name="type">
+					<xsd:complexType>
+						<xsd:choice minOccurs="0" maxOccurs="unbounded">
+							<xsd:element ref="event" />
+							<xsd:element ref="method" />
+							<xsd:element ref="field" />
+							<xsd:element ref="property" />
+							<xsd:element ref="attribute"></xsd:element>
+						</xsd:choice>
+						<xsd:attribute name="name" type="xsd:string" />
+						<xsd:attribute name="feature" />
+						<xsd:attribute name="featurevalue" type="xsd:boolean" />
+						<xsd:attribute name="featuredefault" type="xsd:boolean" />
+					</xsd:complexType>
```

Comment:
I think this should be recursive - technically it's allowed to have nested types within nested types.